apps/extension/src/components/Request/index.tsx

```diff
@@ -0,0 +1,160 @@
+                    {item.description}
+                  </Text>
+                </View>
+              );
+            })}
+          </View>
+        ) : type == requestType.signatureRequest ? (
+          <View className="request__rule mx-[23px] bg-[#01131F] rounded-[8px] border border-solid border-[#01131F] px-[19px] py-[10px] mb-[40px]">
+            <Text
+              style={{ color: '#587A90' }}
+              className="px-[20px] text-center pb-[10px] request__rule-header weight-[400] text-[12px]"
+            >
+              Your signature has been requested
+            </Text>
+            {rules?.map((item, index) => {
+              return (
+                <View className="mb-[5px] request__rule-list flex flex-row gap-2 items-center">
+                  <Image
+                    className="w-[14px] h-[14px]"
+                    source={{ uri: item.icon }}
+                  />
+                  <Text className="weight-[400] text-[14px]">
+                    {item.description}
+                  </Text>
+                </View>
+              );
+            })}
+          </View>
+        ) : (
+          <></>
+        )}
+        <Text
+          style={{ color: '#587A90' }}
+          className="request__notification-message mx-auto mb-[10px] weight-[400px] text-[12px]"
+        >
+          Only connect to websites you trust!
+        </Text>
+        <View className="request__notification-box bg-[#88c3bf1a] flex flex-row px-[12px] py-[12px] mx-[20px] items-center gap-2 mb-[13px]">
+          <Image
+            className="w-[18px] h-[18px]"
+            source={{ uri: '/img/request-screen/exclamation_image.png' }}
+          />
+          <Text
+            style={{ color: '#587A90' }}
+            className="weight-[400px] text-[10px]"
+          >
+            This action does not make any fund transfer. This site cannot
+            transfer fund without your permission
+          </Text>
+        </View>
+        <LinearGradient
+          className="request__button--primary"
+          colors={['#8FC5BE', '#1BA0DA']}
+          style={{
+            flex: 1,
+            width: '90%',
+            margin: 'auto',
+            minHeight: 40,
+            borderRadius: 8,
+            marginBottom: '10px',
+          }}
+        >
+          <TouchableOpacity style={{}} className="" onPress={() => {}}>
+            <Text style={{ textAlign: 'center', lineHeight: 40 }}>
+              {buttonPrimaryContent}
+            </Text>
+          </TouchableOpacity>
+        </LinearGradient>
+        <TouchableOpacity
+          style={{
+            width: '80%',
+            minHeight: 15,
+            margin: 'auto',
+            borderRadius: 8,
+            backgroundColor: 'black',
+          }}
+        >
+          <Text
+            style={{ textAlign: 'center', lineHeight: 15, color: '#88c3c0' }}
+          >
+            Deny
+          </Text>
+        </TouchableOpacity>
+      </View>
+    </View>
```

Comment:
Too complex, conditional check. Split it into smaller child component instead. Keep a component dead simple with less than `120` lines if possible. Debug/refactor a huge component took us more time to keep it small.